packages/data-context/src/sources/VersionsDataSource.ts

```diff
@@ -11,53 +19,116 @@ interface VersionData {
   latest: Version
 }
 
+const REMOTE_MANIFEST_URL = 'https://download.cypress.io/desktop.json'
+const NPM_CYPRESS_REGISTRY = 'https://registry.npmjs.org/cypress'
+
 export class VersionsDataSource {
-  static result: undefined | Promise<string>
+  private _initialLaunch: boolean
+  private _currentTestingType: TestingType | null
+  private _latestVersion: Promise<string>
+  private _npmMetadata: Promise<Record<string, string>>
+
+  constructor (private ctx: DataContext) {
+    this._initialLaunch = true
+    this._currentTestingType = this.ctx.coreData.currentTestingType
+    this._latestVersion = this.getLatestVersion()
+    this._npmMetadata = this.getVersionMetadata()
+  }
 
   /**
    * Returns most recent and current version of Cypress
    * {
    *   current: {
+   *     id: '8.7.0',
    *     version: '8.7.0',
    *     released: '2021-10-15T21:38:59.983Z'
    *   },
    *   latest: {
+   *     id: '8.8.0',
    *     version: '8.8.0',
    *     released: '2021-10-25T21:38:59.983Z'
    *   }
    * }
    */
-  async versions (): Promise<VersionData> {
-    const currentCypressVersion = require('@packages/root')
+  async versionData (): Promise<VersionData> {
+    this.resetLatestVersionTelemetry()
+
+    const latestVersion = await this._latestVersion
+    const npmMetadata = await this._npmMetadata
 
-    VersionsDataSource.result ??= execa(`npm`, [`view`, `cypress`, `time`, `--json`]).then((result) => result.stdout)
+    const latestVersionMetadata: Version = {
+      id: latestVersion,
+      version: latestVersion,
+      released: npmMetadata[latestVersion] ?? new Date().toISOString(),
+    }
 
-    const result = await VersionsDataSource.result
+    this._initialLaunch = false
 
-    const json = JSON.parse(result)
+    return {
+      latest: latestVersionMetadata,
+      current: {
+        id: pkg.version,
+        version: pkg.version,
+        released: npmMetadata[pkg.version] ?? new Date().toISOString(),
+      },
+    }
+  }
+
+  resetLatestVersionTelemetry () {
+    if (this.ctx.coreData.currentTestingType !== this._currentTestingType) {
+      debug('resetting latest version telemetry call due to a different testing type')
+      this._currentTestingType = this.ctx.coreData.currentTestingType
+      this._latestVersion = this.getLatestVersion()
+    }
+  }
 
-    delete json['modified']
-    delete json['created']
+  private async getVersionMetadata (): Promise<Record<string, string>> {
+    const response = await this.ctx.util.fetch(NPM_CYPRESS_REGISTRY)
+    const responseJson = await response.json()
 
-    const latest = Object.keys(json).sort().reverse()?.[0]
+    debug('NPM release dates %o', responseJson.time)
 
-    if (!latest) {
-      throw Error('Could not get npm info for Cypress')
+    return responseJson.time
+  }
+
+  private async getLatestVersion (): Promise<string> {
```

Comment:
We probably want to early-out on this and the `npm` metadata calls if `this.ctx.isRunMode === true`

Reply:
[`f81145c` (#20107)](https://github.com/cypress-io/cypress/pull/20107/commits/f81145c0d9c91e4f26f1d16cfce5e5997053db21)